__init__.py

```diff
@@ -622,9 +679,17 @@ def handle_no_backend_selected(self, message):
                     callback=handle_intent_aborted)
     def handle_stt_menu(self):
         self.state = SetupState.SELECTING_STT
-        self.gui["offlineSTT"] = self.setup.offline_stt_module
-        self.gui["onlineSTT"] = self.setup.online_stt_module
-        self.handle_display_manager("BackendLocalSTT")
+        model = get_stt_lang_configs(lang=self.selected_language, include_dialects=True)
+        supported_stt_engines = list()
+        for engine, configs in model.items():
+            # For Display purposes, we want to show the engine name without the underscore or dash and capitalized all
+            plugin_display_name = engine.replace("_", " ").replace("-", " ").title()
```

Comment:
the configs usually contain a "display_name" key set by the plugin itself, that should be checked and only if not set trigger this logic. While this key is optional in plugins i think all our plugins provide it

Reply:
display_name does not exist for engine name, only the internal configuration entries have a display_name, where is the display_name in this output for example ``` {'ovos-stt-plugin-selene': [{'lang': 'en-AU', 'display_name': 'English (Australia)', 'offline': False}, {'lang': 'en-CA', 'display_name': 'English (Canada)', 'offline': False}, {'lang': 'en-IN', 'display_name': 'English (India)', 'offline': False}, {'lang': 'en-IE', 'display_name': 'English (Ireland)', 'offline': False}, {'lang': 'en-NZ', 'display_name': 'English (New Zealand)', 'offline': False}, {'lang': 'en-PH', 'display_name': 'English (Philippines)', 'offline': False}, {'lang': 'en-ZA', 'display_name': 'English (South Africa)', 'offline': False}, {'lang': 'en-GB', 'display_name': 'English (United Kingdom)', 'offline': False}, {'lang': 'en-US', 'display_name': 'English (United States)', 'offline': False}], 'ovos-stt-plugin-chromium': [{'lang': 'en-AU', 'pfilter': False, 'display_name': 'English (Australia)', 'offline': False}, {'lang': 'en-CA', 'pfilter': False, 'display_name': 'English (Canada)', 'offline': False}, {'lang': 'en-IN', 'pfilter': False, 'display_name': 'English (India)', 'offline': False}, {'lang': 'en-IE', 'pfilter': False, 'display_name': 'English (Ireland)', 'offline': False}, {'lang': 'en-NZ', 'pfilter': False, 'display_name': 'English (New Zealand)', 'offline': False}, {'lang': 'en-PH', 'pfilter': False, 'display_name': 'English (Philippines)', 'offline': False}, {'lang': 'en-ZA', 'pfilter': False, 'display_name': 'English (South Africa)', 'offline': False}, {'lang': 'en-GB', 'pfilter': False, 'display_name': 'English (United Kingdom)', 'offline': False}, {'lang': 'en-US', 'pfilter': False, 'display_name': 'English (United States)', 'offline': False}]} ```

there is only the key "ovos-stt-plugin-selene" from which it builds the engine display_name

Reply:
why do you show plugin_display_name instead of internal display_name? the design here is that each internal entry is displayed, can have any number per plugin, you can include plugin name if you want in UI, but the main display_name is what matters here and is not set below in `supported_stt_engines`

Reply:
note that this only shows alternative models/dialects for the main lang, all these options are relevant for the lang selected in previous step

Reply:
It was originally decided to support only a higher level of language change and ignore the dialects, till there is more support for this

Reply:
that was at core level, this is specific to STT not the same thing, supporting dialects as global language is very different from supporting dialects reported by individual plugins at the plugin level itself. this is about STT accuracy

and this is not only about dialects, install vosk and you can choose large and small models for example